config/v1/types_infrastructure.go

```diff
@@ -98,7 +98,8 @@ type InfrastructureStatus struct {
 	// The 'External' mode indicates that the control plane is hosted externally to the cluster and that
 	// its components are not visible within the cluster.
 	// +kubebuilder:default=HighlyAvailable
-	// +kubebuilder:validation:Enum=HighlyAvailable;SingleReplica;External
+	// +openshift:validation:FeatureGateAwareEnum:featureGate="",enum=HighlyAvailable;SingleReplica;External
+	// +openshift:validation:FeatureGateAwareEnum:featureGate=HighlyAvailableArbiter,enum=HighlyAvailable;HighlyAvailableArbiter;SingleReplica;External
 	ControlPlaneTopology TopologyMode `json:"controlPlaneTopology"`
```

Comment:
This field should not be able to modified day 2 right? The test you have in place shows that it can be changed. Your proposal explicitly calls out that this cannot be changed day 2, so why does that test pass? 

In general, I think this field should never have been mutable, perhaps we just need to make it immutable

Reply:
Ah, this was an older test where we originally had planned to update in the future (i.e moving from arbiter cluster to regular 3Node cluster), but regardless of that, to my knowledge we don't have something that enforces the status of ControlPlaneTopology, so that's why this test passes. 

Changing this status is not supported in the current version of OCP, and I don't know if it would be a problem to make it immutable for the time being, since I think that would be a behavior change that some components might be relying on, for example I don't know if during bootstrap this ever gets applied then changed.

Reply:
I'm going to bring this up at the control plane arch call today and see what we think. IMO, this should always have been marked as immutable (once its set that is, `""` -> "<value>"` should be allowed)